pydis_site/apps/api/viewsets/bot/user.py

```diff
@@ -118,4 +139,75 @@ class UserViewSet(BulkCreateModelMixin, ModelViewSet):
     """
 
     serializer_class = UserSerializer
-    queryset = User.objects
+    queryset = User.objects.all()
+    pagination_class = UserListPagination
+
+    def get_serializer(self, *args, **kwargs) -> ModelSerializer:
+        """Set Serializer many attribute to True if request body contains a list."""
+        if isinstance(kwargs.get('data', {}), list):
+            kwargs['many'] = True
+
+        return super().get_serializer(*args, **kwargs)
+
+    @action(detail=False, methods=["PATCH"], name='user-bulk-patch')
+    def bulk_patch(self, request: Request) -> Response:
+        """
+        Update multiple User objects in a single request.
+
+        ## Route
+        ### PATCH /bot/users/bulk_patch
+        Update all users with the IDs.
+        `id` field is mandatory, rest are optional.
+
+        #### Request body
+        >>> [
+        ...     {
+        ...     'id': int,
+        ...     'name': str,
+        ...     'discriminator': int,
+        ...     'roles': List[int],
+        ...     'in_guild': bool
+        ...     },
+        ...     {
+        ...     'id': int,
+        ...     'name': str,
+        ...     'discriminator': int,
+        ...     'roles': List[int],
+        ...     'in_guild': bool
+        ...     },
+        ... ]
+
+        #### Status codes
+        - 200: Returned on success.
+        - 400: if the request body was invalid, see response body for details.
+        """
+        queryset = self.get_queryset()
+        try:
+            object_ids = [item["id"] for item in request.data]
+        except KeyError:
+            # user ID not provided in request body.
+            resp = {
+                "Error": "User ID not provided."
+            }
+            return Response(resp, status=status.HTTP_400_BAD_REQUEST)
+
+        filtered_instances = queryset.filter(id__in=object_ids)
+
+        if filtered_instances.count() != len(object_ids):
+            # If all user objects passed in request.body are not present in the database.
+            resp = {
+                "Error": "User object not found."
+            }
+            return Response(resp, status=status.HTTP_404_NOT_FOUND)
+
+        serializer = self.get_serializer(
+            instance=filtered_instances,
+            data=request.data,
+            many=True,
+            partial=True
+        )
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Same thing but with less code. `raise_exception` will raise a 400 for invalid data. `Response` defaults to a 200.

```suggestion
        serializer.is_valid(raise_exception=True):
        serializer.save()
        
        return Response(serializer.data)
```